packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts

```diff
@@ -32,12 +32,12 @@ export class HostedPluginDeployerHandler implements PluginDeployerHandler {
     /**
      * Managed plugin metadata backend entries.
      */
-    private currentBackendPluginsMetadata: PluginMetadata[] = [];
+    private readonly currentBackendPluginsMetadata = new Map<string, PluginMetadata>();
```

Comment:
I don't see why we moved from arrays to map and change method names in that file ?
to use a cache ?

Reply:
for fast access in `getDeployedPluginMetadata `

Reply:
map entries are never removed ?

Reply:
It's possible to uninstall plugins in Gitpod, entries are removed as consequence as well. I can add the context menu to the plugin view to uninstall plugins from the vanilla Theia as well. Would it be fine to add as another PR?

Reply:
well I would say that if we populate map with entries, it should have ways to clean it up upstream yes

Reply:
ok, will cover it as part of https://github.com/eclipse-theia/theia/issues/5507